lightning/src/chain/channelmonitor.rs

```diff
@@ -2451,7 +2451,8 @@ impl<Signer: Sign> ChannelMonitorImpl<Signer> {
 					output: outp.clone(),
 				});
 				break;
-			} else if let Some(ref broadcasted_holder_revokable_script) = self.broadcasted_holder_revokable_script {
+			}
+			if let Some(ref broadcasted_holder_revokable_script) = self.broadcasted_holder_revokable_script {
```

Comment:
One step further to minimize unseen interactions like this could be to unset `self.broadcasted_holder_revokeable_script` in `block_disconnected`/`transaction_unconfirmed`'s `onchain_events_awaiting_threshold_conf`. If the removed event is of type `MaturingOutput` and `SpendableOutputDescriptor == DelayedPaymentOutput`, `self.broadcasted_holder_revokeable_script==None` I think ?

Reply:
I mean with this no longer being an `if else` branch list it should "always work", and unsettnig scripts to match seems risky.